src/proxies/clientSDKProxies.ts

```diff
@@ -0,0 +1,23 @@
+import { ClientSDKCache } from '../cache/ClientSDKCache';
+import { ClientSDKContextType } from '../context/clientSDKContext';
+import { ClientSDKProxyContextType } from '../context/clientSDKProxyContext';
+import { wrapInCacheProxy } from './clientSDKCacheProxy';
+import { wrapInLogProxy } from './clientSDKLogProxy';
+
+export function wrapInProxies(
+  client: ClientSDKContextType
+): ClientSDKProxyContextType {
+  const sdkCache = client ? new ClientSDKCache(client) : client;
+
+  return (component: string, cache = false) => {
+    const sdkClient = client ? wrapInLogProxy(client, component) : client;
+
+    if (cache && sdkCache && sdkClient) {
+      sdkCache.swapClient(sdkClient);
```

Comment:
not sure it'll work with multiple wrapped clients
could you please add test, something like
```
const client1 = wrapInProxies(client)('a', true);
const client2 = wrapInProxies(client)('b', true);
// ... and then do a few request from each?
```

Reply:
It should cause instance of cache stored in a closure and will be different for each provider